packages/ef-runtime-client/src/ui/efui.tsx

```diff
@@ -0,0 +1,126 @@
+import { Component } from "preact/compat";
+import { IComponentRegistry } from "../ComponentRegistry";
+import styles from "./styles";
+
+type EFUIProps = { componentRegistry: IComponentRegistry };
+type EFUIState = {
+  showComponents: boolean;
+  needsRefresh: boolean;
+  components: { [key: string]: string };
+};
+
+export default class EFUI extends Component<EFUIProps, EFUIState> {
+  private localOverrides: { [key: string]: string };
+  private componentRegistry: IComponentRegistry;
+
+  constructor(props: EFUIProps) {
+    super(props);
+    this.componentRegistry = props.componentRegistry;
+    this.localOverrides =
+      JSON.parse(localStorage.getItem("ef-overrides")) || {};
+    this.state = {
+      showComponents: false,
+      needsRefresh: false,
+      components: {
+        ...this.componentRegistry.getRegistry(),
+        ...this.localOverrides,
+      },
+    };
```

Comment:
I think I remember you mentioning but you can't use hooks or a functional component here right?

Reply:
I wish I could. I can use functional components but no hooks. I tried importing from "preact", "preact/compat" and "preact/hook" with no luck.